examples/main/README.md

```diff
@@ -4,14 +4,28 @@ This example program allows you to use various LLaMA language models in an easy
 
 ## Table of Contents
 
-1. [Quick Start](#quick-start)
-2. [Common Options](#common-options)
-3. [Input Prompts](#input-prompts)
-4. [Interaction](#interaction)
-5. [Context Management](#context-management)
-6. [Generation Flags](#generation-flags)
-7. [Performance Tuning and Memory Options](#performance-tuning-and-memory-options)
-8. [Additional Options](#additional-options)
+- [llama.cpp/example/main](#llamacppexamplemain)
+  - [Table of Contents](#table-of-contents)
```

Comment:
I'm not the biggest fan of expanding this table of contents. I was trying to keep it small so it wouldn't be overwhelming for people who wanted to look at the docs. I'm guessing this might not have been your intention, but rather the work of some sort of automatic formatter. Either way, I'd recommend leaving this sort of change for a separate pull request.

But if this is really a change people want and you want to put it in this pull request, I'd at least drop these top two lines as they are entirely redundant. One links to the top where you will already be at when you see the table of contents, and the other links to the table of contents!
```suggestion
```